python/ray/serve/_private/router.py

```diff
@@ -350,25 +367,46 @@ def __init__(
         self_actor_id: str,
         handle_source: DeploymentHandleSource,
         event_loop: asyncio.BaseEventLoop,
-        replica_scheduler: Optional[ReplicaScheduler],
         enable_strict_max_ongoing_requests: bool,
+        node_id: str,
+        availability_zone: Optional[str],
+        prefer_local_node_routing: bool,
         resolve_request_arg_func: Coroutine = resolve_deployment_response,
+        request_router_class: Optional[Callable] = None,
+        replica_scheduler: Optional[ReplicaScheduler] = None,
     ):
         """Used to assign requests to downstream replicas for a deployment.
 
         The scheduling behavior is delegated to a ReplicaScheduler; this is a thin
         wrapper that adds metrics and logging.
         """
-
-        self._event_loop = event_loop
+        self._controller_handle = controller_handle
         self.deployment_id = deployment_id
+        self._self_actor_id = self_actor_id
+        self._handle_source = handle_source
+        self._event_loop = event_loop
+        self._request_router_class = request_router_class
         self._enable_strict_max_ongoing_requests = enable_strict_max_ongoing_requests
+        self._node_id = node_id
+        self._availability_zone = availability_zone
+        self._prefer_local_node_routing = prefer_local_node_routing
         # By default, deployment is available unless we receive news
         # otherwise through a long poll broadcast from the controller.
         self._deployment_available = True
 
-        self._replica_scheduler: ReplicaScheduler = replica_scheduler
+        # The request router will be lazy loaded to decouple form the initialization.
+        self._request_router: Optional[ReplicaScheduler] = replica_scheduler
+
+        if self._event_loop.is_running():
+            future = asyncio.run_coroutine_threadsafe(create_event(), self._event_loop)
+            self._request_router_initialized = future.result()
+        else:
+            self._request_router_initialized = asyncio.Event()
```

Comment:
usually `AsyncioRouter` is created in an event loop that's different from the router event loop right? do we not need to make sure the asyncio event is created in the correct event loop